x-pack/plugins/enterprise_search/server/routes/enterprise_search/api_keys.ts

```diff
@@ -32,8 +28,22 @@ export function registerApiKeysRoutes(
     elasticsearchErrorHandler(log, async (context, request, response) => {
       const indexName = decodeURIComponent(request.params.indexName);
       const { keyName } = request.body;
+      const { security: coreSecurity } = await context.core;
 
-      const createResponse = await createApiKey(request, security, indexName, keyName);
+      const createResponse = await coreSecurity.authc.apiKeys.create({
```

Comment:
I would rather do the same changes in the `createApiKey` function as we have some tests there. Where if you change the place what is called rest should be same anyways.

Reply:
@efegurkan Sorry for the delayed response. Addressed in https://github.com/elastic/kibana/pull/189713/commits/883ea865c1109305fc5d2c821bf4bcd2833c6356